Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityCameraProfileInspector.cs

```diff
@@ -57,34 +80,157 @@ public override void OnInspectorGUI()
+
+                        if ((CameraClearFlags)opaqueClearFlags.intValue == CameraClearFlags.Color)
+                        {
+                            opaqueBackgroundColor.colorValue = EditorGUILayout.ColorField("Background Color", opaqueBackgroundColor.colorValue);
+                        }
+
+                        opaqueQualityLevel.intValue = EditorGUILayout.Popup("Quality Setting", opaqueQualityLevel.intValue, QualitySettings.names);
+
+                        EditorGUILayout.Space();
+                        EditorGUILayout.LabelField("Transparent", EditorStyles.boldLabel);
+
+                        EditorGUILayout.PropertyField(transparentNearClip, nearClipTitle);
+                        EditorGUILayout.PropertyField(transparentFarClip, farClipTitle);
+                        EditorGUILayout.PropertyField(transparentClearFlags, clearFlagsTitle);
+
+                        if ((CameraClearFlags)transparentClearFlags.intValue == CameraClearFlags.Color)
+                        {
+                            transparentBackgroundColor.colorValue = EditorGUILayout.ColorField("Background Color", transparentBackgroundColor.colorValue);
+                        }
+
+                        transparentQualityLevel.intValue = EditorGUILayout.Popup("Quality Setting", transparentQualityLevel.intValue, QualitySettings.names);
                     }
+                }, showDisplaySettingsPreferenceKey);
+
+                serializedObject.ApplyModifiedProperties();
+            }
+        }
 
-                    opaqueQualityLevel.intValue = EditorGUILayout.Popup("Quality Setting", opaqueQualityLevel.intValue, QualitySettings.names);
+        private void RenderList(SerializedProperty list)
+        {
+            bool changed = false;
+
+            using (new EditorGUILayout.VerticalScope())
+            {
+                if (list == null || list.arraySize == 0)
+                {
+                    EditorGUILayout.HelpBox("The Mixed Reality Camera System will use default settings.\nAdd a settings provider to customize the camera.", MessageType.Info);
                 }
 
-                EditorGUILayout.Space();
-                EditorGUILayout.LabelField("Transparent Display Settings", EditorStyles.boldLabel);
+                if (InspectorUIUtility.RenderIndentedButton(addSettingsProviderTitle, EditorStyles.miniButton))
                 {
-                    EditorGUILayout.PropertyField(transparentNearClip, nearClipTitle);
-                    EditorGUILayout.PropertyField(transparentFarClip, farClipTitle);
-                    EditorGUILayout.PropertyField(transparentClearFlags, clearFlagsTitle);
+                    list.InsertArrayElementAtIndex(list.arraySize);
+                    SerializedProperty provider = list.GetArrayElementAtIndex(list.arraySize - 1);
+
+                    SerializedProperty providerName = provider.FindPropertyRelative("componentName");
```

Comment:
Another area for creating re-use function. All of the data providers I believe use IMixedRealityServiceConfiguration and could have a base method to fill these properties and then remainder properties filled here by each inspector

Reply:
good idea. can you open an issue to track? i'd like to keep this PR as focused as possible.

Reply:
#6372